sdk/core/Azure.Core/src/TokenRequestContext.cs

```diff
@@ -96,5 +131,26 @@ public TokenRequestContext(string[] scopes, string? parentRequestId = default, s
         /// If you don't handle CAE responses in these API calls, your app could end up in a loop retrying an API call with a token that is still in the returned lifespan of the token but has been revoked due to CAE.
         /// </remarks>
         public bool IsCaeEnabled { get; }
+
+        /// <summary>
+        /// Indicates whether to enable Proof of Possession (PoP) for the requested token.
+        /// </summary>
+        public bool IsProofOfPossessionEnabled { get; }
+
+        /// <summary>
+        /// The nonce value required for PoP token requests. This is typically retrieved from the WWW-Authenticate header of a 401 challenge response.
+        /// This is used in combination with <see cref="ResourceRequestUri"/> and <see cref="ResourceRequestMethod"/> to generate the PoP token.
+        /// </summary>
+        public string? ProofOfPossessionNonce { get; }
+
+        /// <summary>
+        /// The HTTP method of the request. This is used in combination with <see cref="ResourceRequestUri"/> and <see cref="ProofOfPossessionNonce"/> to generate the PoP token.
+        /// </summary>
+        public HttpMethod? ResourceRequestMethod => _method;
```

Comment:
In SCM, we are using `string` instead of the BCL HttpMethod type.  Would it make sense to align our Core libraries around this convention?

Reply:
I don't feel strongly and it didn't come up in arch board. But this type is better for validation purposes, IMO.

Reply:
@KrzysztofCwalina do you have an opinion on this one?

Reply:
I don't have a strong opinion on this, but we have RequestMethod in Azure.Core and it seems like we should use it.